examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/handlers/ApplicationLauncherManagerImpl.java

```diff
@@ -0,0 +1,161 @@
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.util.Log;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.Observer;
+import com.matter.tv.server.tvapp.Application;
+import com.matter.tv.server.tvapp.ApplicationLauncherManager;
+import com.matter.tv.server.tvapp.LauncherResponse;
+import com.matter.tv.server.utils.EndpointsDataStore;
+import com.matter.tv.server.utils.InstallationObserver;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class ApplicationLauncherManagerImpl implements ApplicationLauncherManager {
+
+  private static final String TAG = "ApplicationLauncherService";
+
+  private volatile boolean registered = false;
+  private PackageManager packageManager;
+  private EndpointsDataStore endpointsDataStore;
+
+  /** Hash Map of packageName & Install Status */
+  private Map<String, InstallationObserver.InstallStatus> lastReceivedInstallationStatus =
+      new HashMap<>();
+
+  private LiveData<InstallationObserver.InstallState> installStateLiveData;
+
+  public ApplicationLauncherManagerImpl(Context context) {
+    packageManager = context.getPackageManager();
+    endpointsDataStore = EndpointsDataStore.getInstance(context);
+    registerSelf(context);
+  }
+
+  private final Observer<InstallationObserver.InstallState> installStateObserver =
+      state -> {
+        lastReceivedInstallationStatus.put(state.getAppPackageName(), state.getStatus());
+        switch (state.getStatus()) {
+          case IN_PROGRESS:
+            // Installation is in progress
+            Log.d(TAG, "Installation of " + state.getAppPackageName() + " in progress");
+            break;
+          case SUCCEEDED:
+            // Installation succeeded
+            Log.d(TAG, "Installation of " + state.getAppPackageName() + " succeeded");
+            break;
+          case FAILED:
+            // Installation failed
+            Log.d(TAG, "Installation of " + state.getAppPackageName() + " failed");
+            break;
+        }
+      };
+
+  private void stopObservingInstallations() {
+    if (installStateLiveData != null) {
+      Log.d("InstallationObserver", "Stopped Observing");
+      installStateLiveData.removeObserver(installStateObserver);
+    }
+  }
+
+  public void unregister() {
+    stopObservingInstallations();
+  }
+
+  private void registerSelf(Context context) {
+    if (registered) {
+      Log.i(TAG, "Package update receiver for matter already registered");
+      return;
+    } else {
+      registered = true;
+    }
+    Log.i(TAG, "Registered the matter package updates receiver");
+
+    installStateLiveData = InstallationObserver.installationStates(context);
+    installStateLiveData.observeForever(installStateObserver);
+    Log.d(TAG, "Started Observing package installations");
+  }
+
+  @Override
+  public int[] getCatalogList() {
+    Log.i(TAG, "Get Catalog List");
+    return new int[] {65521, 8891, 3191};
```

Comment:
where did these values come from?

Reply:
I've added them for testing purposes. I can remove or add the ones which are more obvious that are test values.